CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,183 @@
+
+The proposed implementation prohibits the proposer from cancelling their own exchange proposal after it has been approved. This is to prevent Granda Mento from being used as a "free option," where the proposer can decide against the exchange late in the process if the price of the exchange is no longer favorable.
+
+[Initial discussions around Granda Mento](https://forum.celo.org/t/discussion-on-granda-mento-enabling-larger-stablecoin-mints/966) showed concern around providing a "fair" price to ensure that large players do not have an unfair opportunity to "dump" CELO for stable tokens. Some ideas for pricing were considered:
+
+1. Oracle price at the time of exchange proposal.
+   * This is vulnerable to oracle attacks.
+2. The price is included by the exchange proposer, and is required to be within X% of the current oracle price.
+   * This is vulnerable to oracle attacks.
+   * This puts more control in the hands of the exchange proposer.
+3. Oracle price at the time of exchange execution.
+   * This is vulnerable to oracle attacks.
+   * Exchanger doesn't know the price at the proposal time.
+4. TWAP between exchange proposal and exchange execution.
+   * Involves smart contract changes to have the TWAP available on chain.
+   * Exchanger doesn't know the price at the proposal time.
+
+It's desirable for both the exchanger and for the Celo community to know what price will be used for the exchange-- this way, the exchanger knows what they're committing to, and the community can decide if they agree with the price. Approaches (1) and (2) the only options that involve knowledge of the price at the start of the trade. While these are both vulnerable to oracle attacks, the proposed implementation's approver and Governance veto serve as safeguards against an exchange with a manipulated price being executed. [NOTE - LOOKING FOR FEEDBACK ON approach 1 vs. 2].
+
+## Backwards Compatibility
+
+The proposal is largely additive, and includes very small changes to existing contracts. Changes are expected to be entirely backward compatible.
+
+## Test Cases
+N/A
+
+## Implementation
+TBD
+
+## Security Considerations
+
+Implementation Risks:
+* Modifications to logic in `StableToken.sol` around which addresses can mint/burn.
+* `GrandaMento.sol`'s logic around minting/burning stable tokens or spending Reserve funds must be accurate.
+* Exchanges in `GrandaMento.sol` must only be executable if they meet the required criteria.
+
+Operational Risks:
+* The multisig may become compromised or approve exchanges that are not favorable from the community's perspective. Governance must be on the lookout for exchange proposals to veto.
+
+Economic Risks:
+* The stability protocol has never dealt with large mints/burns apart from the initial allocations of cUSD and cEUR. An extremely large mint/burn quantity could potentially affect stability.
```

Comment:
That's an interesting risk. Also if someone's plans with a large amount of cXXX minted through Granda Mento change and they dump them on the market (improbable though? And that could come from any large cXXX holder). Mento might need a while to contract or expand that much supply. Probably a larger Mento discussion